app/models/log.rb

```diff
@@ -343,31 +343,27 @@ def reset_invalidated_dependent_fields!
   PIO = PostcodeService.new
 
   LA_CHANGES = {
-    "E07000027" => "E06000064", # Barrow-in-Furness => Westmorland and Furness
-    "E07000030" => "E06000064", # Eden => Westmorland and Furness
-    "E07000031" => "E06000064", # South Lakeland => Westmorland and Furness
-    "E07000026" => "E06000063", # Allerdale => Cumberland
-    "E07000028" => "E06000063", # Carlisle => Cumberland
-    "E07000029" => "E06000063", # Copeland => Cumberland
-    "E07000163" => "E06000065", # Craven => North Yorkshire
-    "E07000164" => "E06000065", # Hambleton => North Yorkshire
-    "E07000165" => "E06000065", # Harrogate => North Yorkshire
-    "E07000166" => "E06000065", # Richmondshire => North Yorkshire
-    "E07000167" => "E06000065", # Ryedale => North Yorkshire
-    "E07000168" => "E06000065", # Scarborough => North Yorkshire
-    "E07000169" => "E06000065", # Selby => North Yorkshire
-    "E07000187" => "E06000066", # Mendip => Somerset
-    "E07000188" => "E06000066", # Sedgemoor => Somerset
-    "E07000246" => "E06000066", # Somerset West and Taunton => Somerset
-    "E07000189" => "E06000066", # South Somerset => Somerset
+    2025 => {
+      "E08000016" => "E08000038", # Barnsley
+      "E08000019" => "E08000039", # Sheffield
+    },
+  }.freeze
+
+  BACKWARDS_LA_CHANGES = {
+    2024 => {
+      "E08000038" => "E08000016", # Barnsley
+      "E08000039" => "E08000019", # Sheffield
+    },
   }.freeze
 
   def get_inferred_la(postcode)
     result = PIO.lookup(postcode)
     location_code = result[:location_code] if result
-    if LA_CHANGES.key?(location_code) && form.start_date.year >= 2023
-      LA_CHANGES[location_code]
-    elsif !(LA_CHANGES.value?(location_code) && form.start_date.year < 2023)
+    if LA_CHANGES[form.start_date.year]&.key?(location_code)
+      LA_CHANGES[form.start_date.year][location_code]
+    elsif BACKWARDS_LA_CHANGES[form.start_date.year]&.key?(location_code)
```

Comment:
Specify the backwards LA changes explicity, so that it doesn't interfere with the `else` below

Reply:
Need to double check if this is actually backwards compatible tho ⌛

Reply:
Updated the description and this is okay to treat as backwards compatible